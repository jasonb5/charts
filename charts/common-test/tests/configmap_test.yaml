suite: test configmap
values:
  - values/common.yaml
templates:
  - app.yaml
chart:
  appVersion: 0.1.0
tests:
  - it: should pass
    set:
      configmap:
        default:
          data:
            CONFIG: /config/config.txt
            PORT: "{{ .Values.service.ports.default.port }}"

      service:
        ports:
          default:
            port: 80
    asserts:
      - isKind:
          of: ConfigMap
        documentIndex: 0
      - equal:
          path: metadata.name
          value: RELEASE-NAME-common-test
        documentIndex: 0
      - hasDocuments:
          count: 1
