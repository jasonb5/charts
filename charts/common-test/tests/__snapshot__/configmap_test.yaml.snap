should pass:
  1: |
    apiVersion: v1
    data:
      PORT: "80"
    kind: ConfigMap
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: common-test
        app.kubernetes.io/version: 0.1.0
        common/workload: default
        helm.sh/chart: common-test-0.1.0
      name: RELEASE-NAME-common-test-config
      namespace: NAMESPACE
  2: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: common-test
        app.kubernetes.io/version: 0.1.0
        common/workload: default
        helm.sh/chart: common-test-0.1.0
      name: RELEASE-NAME-common-test
      namespace: NAMESPACE
    spec:
      ports:
        - name: common-test
          port: 80
          protocol: TCP
          targetPort: 80
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: common-test
        common/workload: default
